src/com/facebook/buck/jvm/kotlin/KotlincToJarStepFactory.java

```diff
@@ -73,13 +96,54 @@ public void createCompileStep(
             pathToSrcsList,
             ImmutableSortedSet.<Path>naturalOrder()
                 .addAll(
-                    Optional.ofNullable(extraClassPath.apply(buildContext))
+                    Optional.ofNullable(extraClassPath.apply(context))
                         .orElse(ImmutableList.of()))
                 .addAll(declaredClasspathEntries)
                 .build(),
             kotlinc,
             extraArguments,
             filesystem));
+
+    ImmutableSortedSet<Path> javaSourceFiles = ImmutableSortedSet.copyOf(
+        sourceFilePaths
+            .stream()
+            .filter(JAVA_PATH_MATCHER::matches)
+            .collect(Collectors.toSet()));
+
+    // Don't invoke javac if we don't have any java files.
+    if (!javaSourceFiles.isEmpty()) {
```

Comment:
@dsyang no, those are public so that the KotlincToJarStepFactory can override the `getBootClassPath()` method with the additional classpaths similar to how JavacToJarStepFactory overrides this method.